File: calculadora-java/src/main/java/org/example/model/operations/Division.java
package org.example.model.operations;

public class Division implements IOperation {
    private int id;

    public Division() {
        this.id = 4;
    }

    public float operate (float valor1, float valor2) throws ArithmeticException {
        if (valor2 == 0) {
            throw new ArithmeticException();
        } else {
            return valor1 / valor2;
        }
    }

    public int getId() {
        return id;
    }
}

File: calculadora-csharp/src/model/operations/Division.cs
using System;

namespace calculadora_csharp.model.operations;

public class Division : IOperation
{
    public int Id { get; set; } = 4;

    
    public float Operate(float num1, float num2) 
    {
        if (num2 == 0)
        {
            throw new ArithmeticException();
        }
        else
        {
            return num1 / num2;
        }
    }
    
}